### PR TITLE
fix log data race

### PR DESCRIPTION
**Pull request by laizy**

### CODE REVIEW

File: common/log/log.go [hunk 1]
```diff
@@ -85,7 +87,19 @@ func GetGID() uint64 {
 	return n
 }
 
-var Log *Logger
+var globalLogger unsafe.Pointer
```
Comment (Honglei-Cong) [inline common/log/log.go:90]: validate pointer must not be nil after init?

File: common/log/log.go [hunk 2]
```diff
@@ -368,11 +369,20 @@ func InitLog(logLevel int, a ...interface{}) {
...
+
+	logger := New(fileAndStdoutWrite, "", log.LUTC|log.Ldate|log.Lmicroseconds, logLevel, logFile)
+
+	return logger
+}
+
+func InitLog(logLevel int, a ...interface{}) {
+	logger := createLog(logLevel, a...)
+
+	swapGlobalLogger(logger)
```
Comment (Honglei-Cong) [inline common/log/log.go]: It's public.
check if old-logger is nil, close it if not nil?

File: common/log/log.go [hunk 3]
```diff
@@ -85,7 +87,19 @@ func GetGID() uint64 {
 	return n
 }
 
-var Log *Logger
+var globalLogger unsafe.Pointer
+
+func Log() *Logger {
```
Comment (Honglei-Cong) [inline common/log/log.go:92]: just a comment.
this function seems not necessary to be public.

**Comment by Honglei-Cong** [comment]
looks good to me.

File: common/log/log.go (hunk 3 above)
Comment (laizy) [reply common/log/log.go:92]: used in http rpc to set the log level.

File: common/log/log.go (hunk 2 above)
Comment (laizy) [reply common/log/log.go]: done

File: common/log/log.go (hunk 1 above)
Comment (laizy) [reply common/log/log.go:90]: there is a init function to call InitLog, so can gurantee globalLogger will never be nil.

**Review by Honglei-Cong — APPROVED** [review]

**Pull request merged**